src/main/java/io/nats/client/ManualConsumer.java

```diff
@@ -13,6 +13,30 @@
 
 package io.nats.client;
 
+import java.time.Duration;
+
 public interface ManualConsumer extends SimpleConsumer {
```

Comment:
Would be nice to have some general docs on the whole interface. I am not a huge fan of the `ManualConsumer` name to be honest - just an idea of the top of my head, but maybe something like `IterableConsumer` or even have consumer implement `Iterator`?

Not strongly against the current naming, I just find it a little bit awkward.

Reply:
Yeah, I'm did not care for ManualConsumer either. It can't just be Iterator, because since it's endless, I need a way to stop (which is in the superclass) and since this next has a timeout, I can't just make it also implement Java Iterator since it's next does not have a timeout. IterableConsumer is good but not my favorite since Iterator is a well known pattern in java.

I don't even know if Consumer is the best here. For instance the SimpleConsumer is really just a handle/holder/context to allow the dev to stop the consuming since messages are being automatically pushed to the handler they provide. The ManualConsumer is just that plus the next methods.

Naming is hard. Just going to go with IterableConsumer for now

Reply:
Yeah, naming is hard. In go client, for endless iteration, in addition to having `consumer.Consume()` - which takes a callback - I added a `consumer.Messages()` method which works pretty much like `ManualConsumer`:
```go
	it, err := cons.Messages()
	if err != nil {
		log.Fatal(err)
	}
	for {
		msg, err := it.Next()
		if err != nil {
			fmt.Println("next err: ", err)
		}
		msg.Ack()
	}
```